src/System.Memory/tests/Memory/TryGetArray.cs

```diff
@@ -45,5 +45,39 @@ public static void OwnedMemoryTryGetArray()
                 Assert.Equal(array[i], segment.Array[i]);
             }
         }
+        
+        [Fact]
+        public static void TryGetArrayFromEmptyMemory()
+        {
+            int[] array = new int[0];
+            Memory<int> memory = array;
+
+            Assert.True(memory.TryGetArray(out ArraySegment<int> segment));
+            Assert.Same(array, segment.Array);
+            Assert.Equal(0, segment.Array.Length);
+
+            Assert.True(Memory<byte>.Empty.TryGetArray(out ArraySegment<byte> byteSegment));
+        }
+
+        [Fact]
+        public static void TryGetArrayFromEmptyOwnedMemory()
+        {
+            int[] array = new int[0];
+            OwnedMemory<int> owner = new CustomMemoryForTest<int>(array);
+
+            Assert.True(owner.Memory.TryGetArray(out ArraySegment<int> segment));
+            Assert.Same(array, segment.Array);
+            Assert.Equal(0, segment.Array.Length);
+        }
+
+        [Fact]
+        public static void TryGetArrayFromEmptyMemoryAndNegativeIndex()
```

Comment:
rename test name :)

Reply:
Renamed to `TryGetArrayFromEmptyNonRetrievableOwnedMemory`. Let me know if that sounds strange. 😄